element_array_ephys/probe.py

```diff
@@ -259,3 +204,63 @@ class Electrode(dj.Part):
         -> master
         -> ProbeType.Electrode
         """
+
+
+def build_electrode_layouts(
+    probe_type: str,
+    site_count_per_shank: int,
+    col_spacing: float = None,
+    row_spacing: float = None,
+    white_spacing: float = None,
+    col_count_per_shank: int = 1,
+    shank_count: int = 1,
+    shank_spacing: float = None,
+    y_origin="bottom",
+) -> list[dict]:
+
+    """Builds electrode layouts.
+
+    Args:
+        probe_type (str): probe type (e.g., "neuropixels 1.0 - 3A")
+        site_count_per_shank (int): site count per shank.
+        col_spacing (float): (um) horizontal spacing between sites. Defaults to None (single column).
+        row_spacing (float): (um) vertical spacing between columns. Defaults to None (single row).
+        white_spacing (float): (um) offset spacing. Defaults to None.
+        col_count_per_shank (int): number of column per shank. Defaults to 1 (single column).
+        shank_count (int): number of shank. Defaults to 1 (single shank).
+        shank_spacing (float): (um) spacing between shanks. Defaults to None (single shank).
+        y_origin (str): {"bottom", "top"}. y value decrements if "top". Defaults to "bottom".
+    """
+    row_count = int(site_count_per_shank / col_count_per_shank)
+    x_coords = np.tile(
+        np.arange(0, (col_spacing or 1) * col_count_per_shank, (col_spacing or 1)),
+        row_count,
+    )
+    y_coords = np.repeat(np.arange(row_count) * (row_spacing or 1), col_count_per_shank)
+
+    if white_spacing:
+        x_white_spaces = np.tile(
+            [white_spacing, white_spacing, 0, 0], int(row_count / 2)
+        )
+        x_coords = x_coords + x_white_spaces
+
+    shank_cols = np.tile(range(col_count_per_shank), row_count)
+    shank_rows = np.repeat(range(row_count), col_count_per_shank)
+
+    electrode_layouts = [
```

Comment:
Without the old for-loop, you could bump the return up here and remove line 266
```suggestion
    return [
```